docs/src/examples/HyperparameterOptimization.md

```diff
@@ -0,0 +1,425 @@
+# Hyperparameter optimization
+Mateusz Baran
+2024-08-03
+
+## Introduction
+
+This example shows how to automatically select the best values of hyperparameters of optimization procedures such as retraction, vector transport, size of memory in L-BFGS or line search coefficients.
+Hyperparameter optimization relies on the [Optuna](https://optuna.org/) [AkibaSanoYanaseToshihikoTakeruKoyama:2019](@cite) Python library because it is much more advanced than similar Julia projects, offering Bayesian optimization with conditional hyperparameters and early stopping.
+
+## General definitions
+
+Here are some general definitions that you will most likely be able to directly use for your problem without any changes.
+Just remember to install `optuna`, for example using `CondaPkg` Julia library.
+
+``` julia
+using Manifolds, Manopt
+using PythonCall
+using BenchmarkTools
+using LineSearches
+
+# This script requires optuna to be available through PythonCall
+# You can install it for example using
+# using CondaPkg
+# ]conda add optuna
+
+optuna = pyimport("optuna")
+
+norm_inf(M::AbstractManifold, p, X) = norm(X, Inf)
+
+struct TTsuggest_int
```

Comment:
I am not so sure what this struct and the following are, but I suppose they are merely collecting results or values? Maybe that could be mentioned?

Reply:
They are purely technical detail for collecting reported data from a single run.

Reply:
Maybe add exactly this sentence (no phrased a bit nicer maybe) somewhere then :)